runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -0,0 +1,372 @@
+            reconfigureSequencerServers(currentLayout, newLayout, true);
+        } catch (InterruptedException | ExecutionException e) {
+            log.error("Bootstrapping sequencer failed due to exception : ", e);
+        }
+    }
+    
+    /**
+     * Attempts to merge the last 2 segments.
+     *
+     * @param currentLayout Current layout
+     * @return True if merge successful, else False.
+     * @throws CloneNotSupportedException  if layout clne fails.
+     * @throws QuorumUnreachableException  if seal or consensus could not be achieved.
+     * @throws LayoutModificationException if merge not possible in layout.
+     * @throws OutrankedException          if consensus is outranked.
+     */
+    public boolean mergeSegments(Layout currentLayout)
+            throws CloneNotSupportedException, QuorumUnreachableException,
+            LayoutModificationException, OutrankedException {
+
+        currentLayout.setRuntime(runtime);
+        sealEpoch(currentLayout);
+
+        LayoutBuilder layoutBuilder = new LayoutBuilder(currentLayout);
+        Layout newLayout = layoutBuilder
+                .mergePreviousSegment(currentLayout.getSegments().size() - 1)
+                .build();
+        newLayout.setRuntime(runtime);
+
+        attemptConsensus(newLayout);
+
+        try {
+            reconfigureSequencerServers(currentLayout, newLayout, true);
+        } catch (InterruptedException | ExecutionException e) {
+            log.error("mergeSegments: Bootstrapping sequencer failed due to exception : ", e);
+        }
+
+        return true;
+    }
+
+    /**
+     * Runs the layout reconfiguration process.
+     * Seals the layout.
+     * Runs paxos.
+     * During paxos if in recovery mode, we increment rank until committed on same epoch.
+     * If not in recovery, we fail with outrankedException.
+     * The new committed layout is then verified by invalidating the runtime.
+     * Finally we reconfigure the servers (bootstrapping sequencer.)
+     *
+     * @param currentLayout          Layout which needs to be sealed.
+     * @param newLayout              New layout to be committed.
+     * @param forceSequencerRecovery Flag. True if want to force sequencer recovery.
+     */
+    private void runLayoutReconfiguration(Layout currentLayout, Layout newLayout,
+                                          final boolean forceSequencerRecovery)
+            throws QuorumUnreachableException, OutrankedException, InterruptedException,
+            ExecutionException {
+
+        // Seals the incremented epoch (Assumes newLayout epoch = currentLayout epoch + 1).
+        currentLayout.setRuntime(runtime);
+        sealEpoch(currentLayout);
+
+        attemptConsensus(newLayout);
+
+        //TODO: Since sequencer reset is moved after paxos. Make sure the runtime has the latest
+        //TODO: layout view and latest client router epoch. (Use quorum layout fetch.)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)